contracts/modules/staking/StakingPool.sol

```diff
@@ -160,6 +161,11 @@ contract StakingPool is IStakingPool {
     _;
   }
 
+  modifier whenSolvent {
+    require(!insolvent, "StakingPool: All coverage has been claimed. Must create new pool");
```

Comment:
Rebase and use custom error please :pray:
@roxdanila unsure about the terminology used here, might be worth a short brainstorming session to see what would be a better wording.